src/charm.py

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/env python3
+
+import logging
+import random
+import string
+import subprocess
+import typing
+from hashlib import sha256
+from pathlib import Path
+from uuid import uuid4
+
+import yaml
+
+from ops.charm import CharmBase
+from ops.main import main
+from ops.model import ActiveStatus, MaintenanceStatus, WaitingStatus
+
+from oci_image import OCIImageResource, OCIImageResourceError
+from interface_service_mesh import ServiceMeshRequires
+
+try:
+    import bcrypt
+except ImportError:
+    subprocess.check_call(["apt", "update"])
+    subprocess.check_call(["apt", "install", "-y", "python3-bcrypt"])
+    import bcrypt
+
+
+def get_or_set(
```

Comment:
This should be converted to using `StoredState` like here:

https://github.com/canonical/spark-operator/blob/master/src/charm.py#L19

I've discovered that `/run` is not a great place to store things, as it can go away if the pod gets deleted and recreated.